Assets/AppCenter/Editor/AndroidPostBuild.cs

```diff
@@ -0,0 +1,163 @@
+﻿using System.IO;
+using System.Text.RegularExpressions;
+#if UNITY_2018_2_OR_NEWER
+using UnityEditor.Android;
+#endif
+using UnityEditor;
+using UnityEngine;
+
+#if UNITY_2018_2_OR_NEWER
+public class AndroidPostBuild : IPostGenerateGradleAndroidProject
+#else
+public class AndroidPostBuild
+#endif
+{
+    private const string GOOGLE_SERVICES_VERSION = "4.0.1";
+
+    public int callbackOrder { get { return 0; } }
+
+    public void OnPostGenerateGradleAndroidProject(string path)
+    {
+        var settings = AppCenterSettingsContext.SettingsInstance;
+        if (settings.UsePush)
+        {
+            OnAndroidPostBuild(path);
+        }
+    }
+
+    public static void OnAndroidPostBuild(string path)
+    {
+        if (EditorUserBuildSettings.exportAsGoogleAndroidProject)
+        {
+            var gradleFilePath = Path.Combine(path, "build.gradle");
+            // On older versions of Unity, there's a bug where path to project is determined wrong.
+            // Reference: https://issuetracker.unity3d.com/issues/android-ipostgenerategradleandroidproject-dot-onpostgenerategradleandroidproject-returns-incorrect-path-when-exporting-a-project
+            if (!File.Exists(gradleFilePath))
+            {
+                var dirInfo = new DirectoryInfo(path);
+                var dirs = dirInfo.GetDirectories();
+                if (dirs.Length > 0)
+                {
+                    path = dirs[0].FullName;
+                }
+            }
+        }
+
+        if (!MoveGoogleJsonFile(path))
+        {
+            return;
+        }
+        MoveCustomGradleScript(path);
+        if (!InjectFirebaseDependencies(path))
+        {
+            return;
+        }
+        var appFilePath = Path.Combine(path, "unity-android-resources\\build.gradle");
+        SwapGoogleAndJcenter(appFilePath);
+        appFilePath = Path.Combine(path, "build.gradle");
+        SwapGoogleAndJcenter(appFilePath);
+    }
+
+    public static void MoveCustomGradleScript(string pathToBuiltProject)
+    {
+        var appAdditionsFolder = AppCenterSettingsContext.AppCenterPath + "/AppCenter/Plugins/Android/Push/AppAdditions";
+        var sourcePath = Path.Combine(appAdditionsFolder, "appcenterpush.gradle");
+        var destPath = Path.Combine(pathToBuiltProject, "appcenterpush.gradle");
+        File.Copy(sourcePath, destPath, true);
+    }
+
+    public static bool MoveGoogleJsonFile(string pathToBuiltProject)
```

Comment:
May be it would be better to have separate method like `CheckIfGoogleServiceFileExist` which will return bool.
As for `Move` method - there could be some inconsistencies here due to e.g. `MoveCustomGradleScript` is void but the name  pattern is the same. Plus someone may expect not to get `bool` results from methods which names are something like `Move`. Anyway I'm for sticking to consistency here.

Reply:
There's nothing to be broken in the second method. We can just return `true` here I guess or do a pointless check.

Reply:
As for `CheckIfGoogleServiceFileExist` - why create a method for one line, while `File.Exists` method does this job perfectly well? :)

Reply:
I'm ok with this if you would like, this just breaks single responsibility principle for method (method not only move but also check for existence). https://stackoverflow.com/a/33969080 
Again, I'm not pretending that this is a single sort of truth, just discussable :)

Reply:
Can agree with the second point, but as for `Move` - I think checking the file existence is a part of any good "move method" :)